src/libs/SubscriptionUtils.ts

```diff
@@ -30,25 +119,231 @@ Onyx.connect({
     waitForCollectionCallback: true,
 });
 
-let ownerBillingGraceEndPeriod: OnyxEntry<number>;
-Onyx.connect({
-    key: ONYXKEYS.NVP_PRIVATE_OWNER_BILLING_GRACE_PERIOD_END,
-    callback: (value) => (ownerBillingGraceEndPeriod = value),
-});
-
-let amountOwed: OnyxEntry<number>;
-Onyx.connect({
-    key: ONYXKEYS.NVP_PRIVATE_AMOUNT_OWNED,
-    callback: (value) => (amountOwed = value),
-});
-
 let allPolicies: OnyxCollection<Policy>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.POLICY,
     callback: (value) => (allPolicies = value),
     waitForCollectionCallback: true,
 });
 
+/**
+ * @returns The date when the grace period ends.
+ */
+function getOverdueGracePeriodDate(): OnyxEntry<number> {
+    return ownerBillingGraceEndPeriod;
+}
+
+/**
+ * @returns Whether the workspace owner has an overdue grace period.
+ */
+function hasOverdueGracePeriod(): boolean {
+    return !!ownerBillingGraceEndPeriod ?? false;
+}
+
+/**
+ * @returns Whether the workspace owner's grace period is overdue.
+ */
+function hasGracePeriodOverdue(): boolean {
+    return !!ownerBillingGraceEndPeriod && Date.now() < new Date(ownerBillingGraceEndPeriod).getTime();
```

Comment:
Question shouldn't this be > ? That is current date is greater than the grace period date, then return `true` ?

Reply:
Of course, you're right